app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/capture_context.rb

```diff
@@ -0,0 +1,190 @@
+      end
+      process_cpu_counters_rate(compute_summation(cpu_counters))
+
+      mem_gauges = @target.containers.collect do |c|
+        fetch_gauges_data("#{c.name}/#{group_id}/memory/usage")
+      end
+      process_mem_gauges_data(compute_summation(mem_gauges))
+    end
+
+    def hawkular_client
+      @client ||= Hawkular::Metrics::Client.new(
+        hakwular_entrypoint, hakwular_credentials, hakwular_options)
+    end
+
+    def hakwular_entrypoint
+      worker_class = ManageIQ::Providers::Kubernetes:: \
+                     ContainerManager::MetricsCollectorWorker
+      URI::HTTPS.build(
+        :host => @ext_management_system.hostname,
+        :port => worker_class.worker_settings[:hawkular_port],
+        :path => worker_class.worker_settings[:hawkular_path])
+    end
+
+    def hakwular_credentials
+      {:token => @ext_management_system.try(:authentication_token)}
+    end
+
+    def hakwular_options
+      {:tenant     => @tenant,
+       :verify_ssl => @ext_management_system.verify_ssl_mode}
+    end
+
+    def fetch_counters_rate(resource)
+      compute_derivative(fetch_counters_data(resource))
+    end
+
+    def fetch_counters_data(resource)
+      sort_and_normalize(
+        hawkular_client.counters.get_data(
+          resource,
+          :starts         => (@start_time.to_i - @interval) * 1000,
+          :bucketDuration => "#{@interval}s"))
+    rescue SystemCallError => e
+      raise CollectionFailure, e.message
+    end
+
+    def fetch_gauges_data(resource)
+      sort_and_normalize(
+        hawkular_client.gauges.get_data(
+          resource,
+          :starts         => @start_time.to_i * 1000,
+          :bucketDuration => "#{@interval}s"))
+    rescue SystemCallError => e
+      raise CollectionFailure, e.message
+    end
+
+    def process_cpu_counters_rate(counters_rate)
+      total_cpu_time = @node_cores * CPU_NANOSECONDS * @interval
+
+      counters_rate.each do |x|
+        timestamp = Time.at(x['start'] / 1000).utc
+        avg_usage = (x['avg'] * 100.0) / total_cpu_time
+        @ts_values[timestamp]['cpu_usage_rate_average'] = avg_usage
+      end
+    end
+
+    def process_mem_gauges_data(gauges)
+      gauges.each do |x|
+        timestamp = Time.at(x['start'] / 1000).utc
+        avg_usage = (x['avg'] / 1.megabytes) * 100.0 / @node_memory
+        @ts_values[timestamp]['mem_usage_absolute_average'] = avg_usage
+      end
+    end
+
+    def compute_summation(data)
+      ts_data = Hash.new { |h, k| h[k] = [] }
+
+      data.flatten.each { |x| ts_data[x['start']] << x }
+      ts_data.delete_if { |_k, v| v.length != data.length }
+
+      ts_data.keys.sort.map do |k|
+        ts_data[k].inject do |sum, n|
+          # Add min, median, max, percentile95th, etc. if needed
+          {
+            'start'          => k,
+            'end'            => [sum['end'], n['end']].max,
+            'avg'            => sum['avg'] + n['avg'],
```

Comment:
This only appears to be used when collecting group metrics ... so it has to get the metrics for all the containers in a pod.  I'm not exactly sure what the metrics actually look like, however if you take the average metric value for a given time span for each container and sum them all up, you should see the entire load of the containers together ... or, in other words, the entire load of the pod (aka, group).

If you averaged the averages, you'll only see the average usage by the containers in that pod and not how much load the pod has altogether.

That's my interpretation at least.